src/app/dataset-view/additional-components/metadata-component/components/form-components/checkbox-field/checkbox-field.component.ts

```diff
@@ -0,0 +1,15 @@
+import { FormGroup } from "@angular/forms";
+import { ChangeDetectionStrategy, Component, Input } from "@angular/core";
+
+@Component({
+    selector: "app-checkbox-field",
+    templateUrl: "./checkbox-field.component.html",
+    styleUrls: ["./checkbox-field.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CheckboxFieldComponent {
+    @Input() form: FormGroup;
+    @Input() controlName: string;
+    @Input() label: string;
+    @Input() dataTestId?: string;
```

Comment:
Please try reusing these common inputs in all form components in a base abstract class component.
I think if you annotate the base class with `@Directive()` decorator, you won't have any issues with `@Input` visibility in templates.

Reply:
Moved to a base abstract class

Reply:
Fixed